src/NServiceBus.Core/Transports/Msmq/MessagePump.cs

```diff
@@ -146,18 +146,16 @@ public void Start(PushRuntimeSettings limitations)
 
                     await concurrencyLimiter.WaitAsync(cancellationToken).ConfigureAwait(false);
 
+                    var tokenSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationTokenSource.Token);
+
                     var receiveTask = Task.Run(async () =>
                     {
                         try
                         {
-                            await receiveStrategy.ReceiveMessage(inputQueue, errorQueue, pipeline).ConfigureAwait(false);
+                            await receiveStrategy.ReceiveMessage(inputQueue, errorQueue, tokenSource, pipeline).ConfigureAwait(false);
```

Comment:
well if you follow this suggested API:

> It also allows the transport to communicate to the pipeline to abort if possible.

I think we need to be prepared for pipelines to throw an OCE, since that would be the way to cancel the pipeline (also we need to consider that in all exception catching behaviors).